pydis_site/apps/api/models/bot/role.py

```diff
@@ -43,7 +45,14 @@ class Role(ModelReprMixin, models.Model):
         ),
         help_text="The integer value of the permission bitset of this role from Discord."
     )
+    position = models.IntegerField(
+        help_text="The position of the role in the role hierarchy of the Discord Guild."
+    )
 
     def __str__(self):
         """Returns the name of the current role, for display purposes."""
         return self.name
+
+    def __lt__(self, other: Role):
```

Comment:
The interpreter is smart enough to figure out a `>` from a `__lt__` dunder, but for checks including equality, we will indeed need `__eq__`. Roles *should* never have an equal position, though, although there's no strict "this will never happen" in the documentation.

I can implement the `__eq__` to cover `<`, `>`, `==`, `<=` and `>=`, though. I just don't think they are really useful.

Reply:
On further inspection, the `Model` class we inherit from has the following method defined:

```py
    def __eq__(self, other):
        if not isinstance(other, Model):
            return False
        if self._meta.concrete_model != other._meta.concrete_model:
            return False
        my_pk = self.pk
        if my_pk is None:
            return self is other
        return my_pk == other.pk
```

Which allows us to test if two instances of the Model are referring to the same database object. Providing a custom `__eq__` would overwrite that functionality.

Reply:
Fair enough.